test/system/libs/helpers.bash

```diff
@@ -418,8 +418,7 @@ function get_system_id() {
   os_release="$(find_os_release)"
 
   if [[ -z "$os_release" ]]; then
-    echo ""
-    return
+    return 1
   fi
 
   echo $(awk -F= '/ID/ {print $2}' $os_release | head -n 1)
```

Comment:
This isn't about this specific change, but a more general comment.

This has always looked needlessly complicated to me.  eg., in the POSIX shell implementation, we simply sourced `os-release` file in a sub-shell and echoed the specific variable we want:
```
get_host_id()
(
    # shellcheck disable=SC1091                                                 
    . /usr/lib/os-release
    echo "$ID"
)

get_host_variant_id()
(
    # shellcheck disable=SC1091                                                 
    . /usr/lib/os-release
    echo "$VARIANT_ID"
)

get_host_version_id()
(
    # shellcheck disable=SC1091                                                 
    . /usr/lib/os-release
    echo "$VERSION_ID"
)
```